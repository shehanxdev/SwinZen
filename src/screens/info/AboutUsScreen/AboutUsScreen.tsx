import React from 'react';
import { View } from 'react-native';

import { Text } from '@sz/components';
import { tw } from '@sz/config';
import { AboutUsPoints, Color, TextAlignment, TextVariant } from '@sz/constants';

import { BaseInfoScreen } from '../components';

export function AboutUsScreen() {
  return (
    <BaseInfoScreen testID="AboutUsScreenTestID">
      <View style={tw`flex-1 bg-[${Color.Primary.Sz900}]/46 rounded-2.5 mx-4 mt-6.25 mb-24 p-6`}>
        <Text variant={TextVariant.Body2Regular} textAlign={TextAlignment.Auto}>
          SwingZen is a new app technology utilizing a cutting-edge biomechanical feedback analytics system to track the
          range of motion of a golferâ€™s body, club, and swing dynamics from any smartphone. No expensive doppler or
          sensors needed!
        </Text>
        {AboutUsPoints.map((data, index) => (
          <View key={index} style={tw`flex-row gap-2 mt-4 ml-3`}>
            <Text variant={TextVariant.Body2Regular} textAlign={TextAlignment.Auto}>
              {`\u2022`}
            </Text>
            <Text variant={TextVariant.Body2Regular} textAlign={TextAlignment.Auto}>
              {data}
            </Text>
          </View>
        ))}
      </View>
    </BaseInfoScreen>
  );
}
