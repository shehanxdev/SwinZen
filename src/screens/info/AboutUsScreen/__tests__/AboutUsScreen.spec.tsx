import { render } from '@testing-library/react-native';
import * as React from 'react';

import { AboutUsScreen } from '../AboutUsScreen';

describe('About Us Screen Screen', () => {
  const testID = 'AboutUsScreenTestID';

  const getRenderedScreen = () => render(<AboutUsScreen />);

  it(`should render About Us Screen correctly`, () => {
    const renderer = getRenderedScreen();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });

  it('should find the About Us Screen via testID', () => {
    const { getByTestId } = getRenderedScreen();
    const foundScreen = getByTestId(testID);
    expect(foundScreen).toBeTruthy();
  });
});
