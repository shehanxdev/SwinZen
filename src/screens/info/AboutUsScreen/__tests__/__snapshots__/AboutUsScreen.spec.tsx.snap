// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Us Screen Screen should render About Us Screen correctly 1`] = `
<BVLinearGradient
  colors={
    [
      4279917603,
      4278984207,
      4278650631,
    ]
  }
  endPoint={
    {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    {
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "position": "relative",
    }
  }
  testID="AboutUsScreenTestID"
>
  <Image
    resizeMode="repeat"
    source={
      {
        "testUri": "../../../src/assets/images/grassBackground.jpg",
      }
    }
    style={
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
        "opacity": 0.1,
      }
    }
  />
  <RCTSafeAreaView
    style={
      {
        "height": "100%",
        "paddingTop": 50,
        "position": "absolute",
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        [
          {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          {
            "flexGrow": 1,
          }
        }
        keyboardShouldPersistTaps="handled"
      >
        <View>
          <View
            style={
              {
                "backgroundColor": "rgba(7, 7, 7, 0.46)",
                "borderRadius": 10,
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
                "marginBottom": 96,
                "marginLeft": 16,
                "marginRight": 16,
                "marginTop": 25,
                "paddingBottom": 24,
                "paddingLeft": 24,
                "paddingRight": 24,
                "paddingTop": 24,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#F8F9FA",
                    "lineHeight": 24,
                  },
                  {
                    "fontFamily": "SourceSansPro-Regular",
                    "fontSize": 16,
                    "fontWeight": "normal",
                    "textAlign": "auto",
                  },
                ]
              }
            >
              SwingZen is a new app technology utilizing a cutting-edge biomechanical feedback analytics system to track the range of motion of a golfer’s body, club, and swing dynamics from any smartphone. No expensive doppler or sensors needed!
            </Text>
            <View
              style={
                {
                  "flexDirection": "row",
                  "gap": 8,
                  "marginLeft": 12,
                  "marginTop": 16,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#F8F9FA",
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 16,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                •
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#F8F9FA",
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 16,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Utilizes cutting edge biomechanical feedback system.
              </Text>
            </View>
            <View
              style={
                {
                  "flexDirection": "row",
                  "gap": 8,
                  "marginLeft": 12,
                  "marginTop": 16,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#F8F9FA",
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 16,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                •
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#F8F9FA",
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 16,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Tracks up to 20 points of body and club for quick frame by frame review.
              </Text>
            </View>
            <View
              style={
                {
                  "flexDirection": "row",
                  "gap": 8,
                  "marginLeft": 12,
                  "marginTop": 16,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#F8F9FA",
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 16,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                •
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#F8F9FA",
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 16,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Delivers data points to your analyzed video such as angles and point movement during swing.
              </Text>
            </View>
            <View
              style={
                {
                  "flexDirection": "row",
                  "gap": 8,
                  "marginLeft": 12,
                  "marginTop": 16,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#F8F9FA",
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 16,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                •
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#F8F9FA",
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 16,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Requires no expensive sensors or doppler system. Only your smartphone.
              </Text>
            </View>
            <View
              style={
                {
                  "flexDirection": "row",
                  "gap": 8,
                  "marginLeft": 12,
                  "marginTop": 16,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#F8F9FA",
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 16,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                •
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#F8F9FA",
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 16,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Delivers exclusive video and swing fault instruction content under the Instruction tab.
              </Text>
            </View>
            <View
              style={
                {
                  "flexDirection": "row",
                  "gap": 8,
                  "marginLeft": 12,
                  "marginTop": 16,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#F8F9FA",
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 16,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                •
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#F8F9FA",
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 16,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                If desired, you can work with an instructor or coach that will have instant access to your information through their instructors.
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
